src/geocat/viz/contourf/contourf.py

```diff
@@ -1,81 +1,435 @@
+
+            Y (:class:`xarray.core.dataarray.DataArray'>` or :class:`numpy.ndarray`): The Y axis data for the dataset. To be specified if not inferred correctly automatically.
+        """
+
+        # Set default flevels, clevels, and colormap
         self._default_cmap = 'coolwarm'
         self._default_flevels = 5
         self._default_clevels = 7
 
-        # Pull out args
+        # Pull out args from data
         self.data = args[0]
 
-        # if in xarray, format as numpy array
+        # If xarray file, format as Numpy array
         if isinstance(self.data, xr.DataArray):
+            self.orig = self.data
             self.data = self.data.values
 
-        # Read in or calculate filled levels
-        if kwargs.get('flevels') is not 'auto':
-            # levels defined by kwargs
-            self.levels = kwargs.get('flevels')
-        elif kwargs.get('flevels') is not None:
-            # take a guess at filled levels
-            self._estimate_flevels
-
-        # Read in or calculate contour levels
-        if kwargs.get('clevels') is not 'auto':
-            # levels defined by kwargs
-            self.levels = kwargs.get('clevels')
-        elif kwargs.get('clevels') is not None:
-            # take a guess at filled levels
-            self._estimate_clevels
-
-        # Pull out child-class specific kwargs
+        # Read in or calculate filled levels using built in function
+        if kwargs.get('contour_fill') is not False:
+            if kwargs.get('flevels') is not None:
+                # levels defined by kwargs
+                self.levels = kwargs.get('flevels')
+            elif kwargs.get('flevels') is None:
+                # take a guess at filled levels
+                self._estimate_flevels
+
+        # Pull in X and Y axis data if specified
+        if kwargs.get("X") is not None:
+            self.X = kwargs.get("X")
+            if kwargs.get("Y") is None:
+                raise AttributeError("If X is defined, Y must also be defined.")
+        else:
+            self.X = kwargs.get("X")
+
+        if kwargs.get("Y") is not None:
+            self.Y = kwargs.get("Y")
+            if kwargs.get("X") is None:
+                raise AttributeError("If Y is defined, X must also be defined.")
+        else:
+            self.Y = kwargs.get("Y")
+
+        # Pull in whether to draw filled contours and/or contour lines
+        self.contour_lines = kwargs.get('contour_lines')
+        self.contour_fill = kwargs.get('contour_fill')
+
+        # Read in or calculate contour levels using built in function
+        if kwargs.get('contour_lines') is not False:
+            if kwargs.get('clevels') is not None:
+                # levels defined by kwargs
+                self.levels = kwargs.get('clevels')
+            elif kwargs.get('clevels') is None:
+                # take a guess at filled levels
+                self._estimate_clevels
```

Comment:
```suggestion
                self._estimate_clevels()
```
Same comment as above